### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

-  elasticurl uses "h2;http/1.1" alpn string by default
        --http2 and --http1.1 flags force "h2" or "http/1.1" and error if they don't get it
-  add h2 tests to integration-testing/http_client_test.py
-  add add_net_test_case(tls_download_medium_file_http2)

### CODE REVIEW

File: integration-testing/http_client_test.py [hunk 1]
```diff
@@ -77,10 +77,27 @@ def test_simple_post(self):
     def test_simple_download(self):
```
Comment (graebm) [inline integration-testing/http_client_test.py:77]: let's make the existing tests be explicitly "_h1" tests
modify the test names and pass "--http1.1"

File: bin/elasticurl/main.c [hunk 2]
```diff
@@ -453,6 +461,10 @@ static void s_on_signing_complete(struct aws_http_message *request, int error_co
 
 static void s_on_client_connection_setup(struct aws_http_connection *connection, int error_code, void *user_data) {
     struct elasticurl_ctx *app_ctx = user_data;
+    if (aws_http_connection_get_version(connection) != app_ctx->required_http_version) {
+        fprintf(stderr, "Error. Connection we got is different from the required version.");
```
Comment (JonathanHenson) [inline bin/elasticurl/main.c:465]: Error. The requested http version, %s, is not supported by the peer., ctx->alpn

File: bin/elasticurl/main.c [hunk 3]
```diff
@@ -453,6 +461,10 @@ static void s_on_signing_complete(struct aws_http_message *request, int error_co
 
 static void s_on_client_connection_setup(struct aws_http_connection *connection, int error_code, void *user_data) {
     struct elasticurl_ctx *app_ctx = user_data;
+    if (aws_http_connection_get_version(connection) != app_ctx->required_http_version) {
```
Comment (JonathanHenson) [inline bin/elasticurl/main.c:464]: how does required_http_version get set if user doesn't specify one explicitly?

Comment (TingDaoK) [reply bin/elasticurl/main.c]: I forgot about this case.🤦‍ Fixed.

**Pull request merged**